src/targets/maven.ts

```diff
@@ -205,6 +206,15 @@ export class MavenTarget extends BaseTarget {
     );
   }
 
+  /**
+   * Deletes the user's `gradle.properties` file.
+   */
+  public deleteUserGradlePropsFile(): Promise<void> {
+    return fsPromises.unlink(
+      join(this.getGradleHomeDir(), GRADLE_PROPERTIES_FILENAME)
```

Comment:
Who creates this file? If it is the `closeAndReleaseRepository` script, then shouldn't this be its responsibility (to remove the file I mean)

Reply:
Then at this point, I'd convert this into something like `withTempDir` with auto-clean up even in case of exceptions for safety. This is much like a [Python Context Manager](https://realpython.com/python-with-statement/#using-the-python-with-statement)